packages/app/src/components/actions/deposit.tsx

```diff
@@ -59,39 +54,21 @@ const Deposit: FC<Props> = ({ name, symbol, tokenBalance, tokenAllowance }) => {
             Current Balance: {`${utils.formatEther(tokenBalance?.toString() || "0")} ${symbol}`}
           </Form.Text>
         </Form.Group>
-
-        {!depositInput || BigNumber.from(tokenAllowance).gte(utils.parseEther(depositInput || "0")) ? (
+        <div className="d-grid gap-2">
+          <ApproveToken
+            symbol={symbol}
+            spender={addresses[name].controller}
+            tokenAddress={contracts[name].token}
+            hasAllowance={isTokenApproved}
+          />
           <Button
             disabled={!depositInput || depositInput.toString() === "0" || depositTx.status === "Mining"}
```

Comment:
Need to also add `!isTokenApproved` to the disabled conditions 

![image](https://user-images.githubusercontent.com/22256858/124475648-3d6e2080-dda2-11eb-9351-23f787726538.png)

Reply:
👍